enclave_build/src/cache.rs

```diff
@@ -0,0 +1,782 @@
+            .ok_or(Error::HashMissing)?;
+
+        // Add algorithm prefix to the hash
+        let image_hash = format!("sha256:{}", hash);
+
+        // Fetch the config JSON string from cache
+        let config_json = self.fetch_config(&image_name)?;
+
+        Ok(ImageDetails::new(
+            image::build_image_reference(&image_name)
+                .map_err(Error::FetchImageDetailsRef)?
+                .whole(),
+            image_hash,
+            image::deserialize_from_reader(config_json.as_bytes())
+                .map_err(Error::FetchImageDetailsSerde)?,
+        ))
+    }
+
+    /// Returns the manifest JSON string from the cache.
+    pub fn fetch_manifest<S: AsRef<str>>(&self, image_name: S) -> Result<String> {
+        let target_path = self.get_image_folder_path(&image_name)?;
+
+        // Read the JSON string from the manifest file
+        let manifest_path = target_path.join(CACHE_MANIFEST_FILE_NAME);
+        let mut manifest_json = String::new();
+        File::open(&manifest_path)
+            .map_err(Error::FetchManifest)?
+            .read_to_string(&mut manifest_json)
+            .map_err(Error::FetchManifest)?;
+
+        if manifest_json.is_empty() {
+            return Err(Error::ManifestEmpty);
+        }
+
+        Ok(manifest_json)
+    }
+
+    /// Returns the config JSON string from the cache.
+    pub fn fetch_config<S: AsRef<str>>(&self, image_name: S) -> Result<String> {
+        let target_path = self.get_image_folder_path(&image_name)?;
+
+        let mut config_json = String::new();
+        File::open(target_path.join(CACHE_CONFIG_FILE_NAME))
+            .map_err(Error::FetchConfig)?
+            .read_to_string(&mut config_json)
+            .map_err(Error::FetchConfig)?;
+
+        if config_json.is_empty() {
+            return Err(Error::ConfigEmpty);
+        }
+
+        Ok(config_json)
+    }
+
+    /// Validates that the image layers are cached correctly by checking them with the layer descriptors
+    /// from the image manifest.
+    fn validate_layers<P: AsRef<Path>>(&self, layers_path: P, manifest_str: &str) -> Result<()> {
+        let manifest_obj: serde_json::Value = serde_json::from_str(manifest_str)
+            .map_err(|_| Error::ImageNotCached("Manifest serialization failed".to_string()))?;
+
+        // Try to get the layer list from the manifest JSON
+        let layers_vec: Vec<serde_json::Value> = manifest_obj
+            .get("layers")
+            .ok_or_else(|| {
+                Error::ImageNotCached("'layers' field missing from manifest JSON.".to_string())
+            })?
+            .as_array()
+            .ok_or_else(|| Error::ImageNotCached("Manifest deserialize error.".to_string()))?
+            .to_vec();
+
+        // Get the cached layers as a HashMap mapping a layer digest to the corresponding layer file
+        let mut cached_layers: HashMap<String, File> = HashMap::new();
+
+        fs::read_dir(layers_path)
```

Comment:
Wow 🥺 I need a fresh mind to review this, but I see some undocumented unwraps :(.

Reply:
Before each unwrap i am doing a .filter in order to keep only the Ok() variants

Reply:
Simplified map creation with a `filer_map` and got rid of unwraps.